jwst/datamodels/tests/test_models.py

```diff
@@ -691,3 +692,10 @@ def test_ifuimage():
 def test_datamodel_raises_filenotfound():
     with pytest.raises(FileNotFoundError):
         DataModel(init='file_does_not_exist.fits')
+
+def test_abvega_offset_model():
+    model = ABVegaOffsetModel('nircam_abvega_offset.asdf')
+    assert isinstance(model, ABVegaOffsetModel)
+    assert hasattr(model, 'abvega_offset')
+    assert isinstance(model.abvega_offset, Table)
+    assert model.abvega_offset.colnames == ['filter', 'pupil', 'abvega_offset']
```

Comment:
Maybe a `model.validate()` call would be good in the test.  Or better, make those required fields in the schema?  @nden I recall there being some `.validate()` methods you have written that do something like this.

Reply:
`ReferenceFileeModel` has its own `validate()` method which checks for the presence of the common required  reference file keywords. When possible I also add `validate` on specific models to make sure they are consistent and that fields I know should be there are really present, for example models in wcs_ref_models.py.

Reply:
I've added a custom `validate()` method to check that the expected columns for a given instrument are in the table.

Reply:
Is there a reason to write a `validate()` method instead of using `required` in the schema directly?  We do use `required` often elsewhere.  An example from our transforms:

https://github.com/spacetelescope/jwst/blob/95b306c76a21a67816cc15407445a5448e2fd0e9/jwst/transforms/schemas/stsci.edu/jwst_pipeline/coords-0.7.0.yaml#L39

(I'm asking mostly for my own education here.)

Reply:
I think `required` would work only for the `abvega_offset` column, because that's the only column that appears in every schema.  The other columns depend on the instrument.  I'm not a schema expert, so please let me know if there is a way to use `required` in combination with `anyOf` in `properties`.